Source/WebCore/bindings/js/JSNodeListCustom.cpp

```diff
@@ -37,6 +37,80 @@
 namespace WebCore {
 using namespace JSC;
 
+const ClassInfo JSStaticNodeList::s_info = {
+    "NodeList"_s,
+    &Base::s_info,
+    nullptr,
+    nullptr,
+    std::nullopt,
+    {
+        &JSStaticNodeList::destroy,
+        &JSStaticNodeList::getCallData,
+        &JSStaticNodeList::getConstructData,
+        &JSStaticNodeList::put,
+        &JSStaticNodeList::putByIndex,
+        &JSStaticNodeList::deleteProperty,
+        &JSStaticNodeList::deletePropertyByIndex,
+        &Base::Base::getOwnPropertySlot,
+        &Base::Base::getOwnPropertySlotByIndex,
+        &JSStaticNodeList::toThis,
+        &Base::Base::getOwnPropertyNames,
+        &Base::Base::getOwnSpecialPropertyNames,
+        &JSStaticNodeList::customHasInstance,
+        &JSStaticNodeList::defineOwnProperty,
+        &JSStaticNodeList::preventExtensions,
+        &JSStaticNodeList::isExtensible,
+        &JSStaticNodeList::setPrototype,
+        &JSStaticNodeList::getPrototype,
+        &JSStaticNodeList::dumpToStream,
+        &JSStaticNodeList::analyzeHeap,
+        &JSStaticNodeList::estimatedSize,
+        &JSStaticNodeList::visitChildren,
+        &JSStaticNodeList::visitChildren,
+        &JSStaticNodeList::visitOutputConstraints,
+        &JSStaticNodeList::visitOutputConstraints,
+    },
```

Comment:
CREATE_METHOD_TABLE should work? If not, probably we should rearchitect the design of hierarchy of `JSNodeList` and `JSStaticNodeList`. (inserting `JSNodeListBase` which implements shared methods for both).

Reply:
`CREATE_METHOD_TABLE` doesn't work as it expands to `&JSStaticNodeList::foo` for all methods, yet we need original `getOwnPropertySlot` and friends to be fast.

Rearchitecting inheritance chain into `JSNodeListBase` / `JSStaticNodeList` / `JSLiveNodeList` is indeed what we should strive for, but currently that would amount to massive changes of `CodeGeneratorJS.pm`, unfortunately.

Reply:
Let's add FIXME since it is not ideal.

Reply:
Added.